packages/magner/lib/utils/core/mixed-check.ts

```diff
@@ -53,6 +53,10 @@ export const updateFieldValues = (
   form: Record<string, any>,
   force?: Record<'readOnly' | 'disabled' | 'hidden', boolean | MixedChecker>,
 ) => {
+  // Save pure disabled function
+  if (typeof field.props.disabledCondition !== 'function') {
+    field.props.disabledCondition = field.props.disabled;
+  }
   const disabled = force?.disabled ?? field.props.disabledCondition;
   if (typeof disabled === 'function') {
     field.props.disabled = disabled.bind(null, {
```

Comment:
I believe that the last line here `field.props.disabled = disabled.bind` can be changed to `field.props.disabledCondition = field.props.disabled.bind`. In this case, we could remove the saving pure functions code lines. `disabled` prop would stay the same as the user writes it, and only the `disabledCondition` will change.

Reply:
@VanishMax Yes, but then you will have to change the logic in the components...